src/codegen/compile.ml

```diff
@@ -9459,20 +9470,32 @@ and compile_dec env pre_ae how v2en dec : VarEnv.t * G.t * (VarEnv.t -> scope_wr
     assert AllocHow.(match M.find_opt name how with
                      | Some (LocalMut _ | StoreHeap | StoreStatic) -> true
                      | _ -> false);
-      let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
+    let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
 
-      ( pre_ae1,
-        alloc_code,
-        (fun ae -> let sr, code = Var.set_val env ae name in compile_exp_as env ae sr e ^^ code),
-        unmodified
-      )
+    ( pre_ae1,
+      alloc_code,
+      (fun ae -> let sr, code = Var.set_val env ae name in compile_exp_as env ae sr e ^^ code),
+      unmodified
+    )
+  | RefD (name, _, e) ->
+    assert AllocHow.(M.find_opt name how = Some (StoreHeap));
+    let pre_ae1, alloc_code = AllocHow.add_local env pre_ae how name in
+
+    ( pre_ae1,
+      alloc_code,
```

Comment:
This may be a problem. You might want to allocate differently than for VarD (a new local, but not a new box). Didn't check closely though.

Reply:
Right, `alloc_code` is gone now.